equinox/nn/_batch_norm.py

```diff
@@ -202,8 +259,15 @@ def _stats(y):
                 norm_var = zero_frac * batch_var + running_var
                 norm_var = (1.0 - warmup_frac) * batch_var + warmup_frac * norm_var
             else:
+                axis_size = jax.lax.psum(jnp.array(1.0), self.axis_name)
```

Comment:
I'm using this to get the length of the "batch" axis - but not sure it's the best / correct way

Reply:
I think this is the correct way! IIRC `psum(1)` is actually special-cased for this purpose.